### PR TITLE
[REVIEW] Add Doxygen style comments to Join CUDA functions

### PR DESCRIPTION
**Pull request by jeanp413**
Doxygen style comments moved from old libgdf repo https://github.com/rapidsai/libgdf/pull/156

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by harrism** [comment]
add to whitelist

**Comment by harrism** [comment]
@jeanp413 Can you add a line to the CHANGELOG.md for this PR? (under 0.5.0 #improvements section).

**Comment by harrism** [comment]
@williamBlazing @jeanp413 can you get this PR synced up with branch-0.6 and get tests to pass?

File: cpp/include/cudf/functions.h [hunk 1]
```diff
@@ -63,20 +63,66 @@ gdf_error gdf_nvtx_range_pop();
...
  */
-/* ----------------------------------------------------------------------------*/
+/* --------------------------------------------------------------------------*/
 gdf_error gdf_count_nonzero_mask(gdf_valid_type const *masks,
                                  gdf_size_type num_rows, gdf_size_type *count);
 
+
 /* column operations */
 
+/* --------------------------------------------------------------------------*/
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: I know something like this made it into the template, but there shouldn't be 3 separate comment blocks for each block comment. Please remove the superfluous `/*----...---*/` from the start and end of each comment block.

File: cpp/include/cudf/functions.h [hunk 2]
```diff
@@ -820,114 +2534,345 @@ gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
...
-                                                                  //(2) for possible types bigger than double, in the future;
-                                gdf_context*        ctxt);        //context info
-
-gdf_error gdf_quantile_aprrox(	gdf_column*  col_in,       //input column with 0 null_count otherwise GDF_VALIDITY_UNSUPPORTED is returned
-                                double       q,            //requested quantile in [0,1]
-                                void*        t_erased_res, //type-erased result of same type as column;
-                                gdf_context* ctxt);        //context info
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief  Performs SQL like WHERE (Filtering)
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: I think this is a good brief comment, but documentation should also include (after the brief) detailed explanation of the functions semantics, expected use, etc. Perhaps that is a task for a further PR. An issue should be filed if so.

File: cpp/include/cudf/functions.h [hunk 3]
```diff
@@ -820,114 +2534,345 @@ gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
...
+ *
+ * @param[in] input column
+ * @param[in] requested quantile in [0,1]
+ * @param[out] result; type-erased result of same type as column;
+ * @param[in] struct with additional info
+  *
+* @Returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_quantile_aprrox(gdf_column* col_in,
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: Is this a typo? The function name should be `gdf_quantile_approx` (two 'p's, one 'r', as in the word approximation).

File: cpp/include/cudf/functions.h [hunk 4]
```diff
@@ -87,112 +133,501 @@ gdf_error gdf_column_free(gdf_column *column);
  * @param[out] output A column whose buffers are already allocated that will 
  * @param[in] columns_to_conat[] The columns to concatenate
  * @param[in] num_columns The number of columns to concatenate
-  * contain the concatenation of the input columns
+ * contain the concatenation of the input columns
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: This line is misplaced. It should be after line 133 (and indented).

File: cpp/include/cudf/functions.h (hunk 2 above)
Comment (williamBlazing) [reply cpp/include/cudf/functions.h]: I will create an issue in that case. We did not create this function, nor do we use it, so we would not be the best for providing a detailed explanation of the functions semantics, expected use, etc

**Pull request merged**